src/desktop/DesktopMonkeyPatch.js

```diff
@@ -1,12 +1,27 @@
 // @flow
 import chalk from "chalk"
+import fs from 'fs-extra'
+import path from 'path'
+import {app} from 'electron'
 import {execSync} from 'child_process'
 import {last} from '../../src/api/common/utils/ArrayUtils'
 import {neverNull} from "../api/common/utils/Utils"
 import {Logger, replaceNativeLogger} from "../api/common/Logger"
 
 const logger = new Logger()
 replaceNativeLogger(global, logger, true)
+process.on('exit', () => {
+	const logDir = path.join(app.getPath('userData'), 'logs')
+	const logFilePath = path.join(logDir, "tutanota_desktop.log")
+	const oldLogFilePath = path.join(logDir, "tutanota_desktop_old.log")
+	const entries = logger.getEntries()
+	fs.mkdirp(logDir)
+	  .then(() => fs.renameSync(logFilePath, oldLogFilePath))
```

Comment:
> Listener functions **must** only perform **synchronous** operations. The Node.js process will exit immediately after calling the 'exit' event listeners causing any additional work still queued in the event loop to be abandoned.

https://nodejs.org/api/process.html#process_event_exit

You have a mix of `renameSync` but it's inside `then` which makes no sense.

I checked and `before-exit` can be prolonged with new async operations but it is not invoked in certain cases.

I think the right thing to do is to replace all async operations where (which is only `mkdrip`)